### PR TITLE
Flag that a peer was bootstrapped via UTXO set snapshot 

### PR DESCRIPTION
**Pull request by kushti**
In this PR:

* outdated < 4.0.100 peer filters removed (as connections with nodes < 4.0.100 not possible after  #2021 
* a flag indicated that a peer was bootstrapped via  UTXO set snapshot added
* the node now is downloading blocks and headers only from peers having all the blocks and headers (will be enhanced after https://github.com/ergoplatform/ergo/issues/1915 implementation)

### CODE REVIEW

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: src/main/scala/org/ergoplatform/network/ModePeerFeature.scala [hunk 1]
```diff
@@ -15,36 +15,58 @@ import scorex.util.serialization.{Reader, Writer}
...
 }
 
 object ModePeerFeature {
 
   import io.circe.syntax._
 
+  val NiPoPoWNormalFlag = 1
+
+  val AllBlocksKept = -1
+  val UTXOSetBootstrapped = -2
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/network/ModePeerFeature.scala:50]: fixed

File: src/main/scala/org/ergoplatform/network/ModePeerFeature.scala [hunk 2]
```diff
@@ -15,36 +15,58 @@ import scorex.util.serialization.{Reader, Writer}
   *
   * @param stateType - information on whether UTXO set is store (so state type is UTXO/Digest)
   * @param verifyingTransactions - whether the peer is verifying transactions
-  * @param nipopowSuffix - whether the peer has has bootstrapped via Nipopows, and length of proof suffix
+  * @param nipopowBootstrapped - whether the peer has bootstrapped via Nipopows
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/network/ModePeerFeature.scala:18]: fixed

File: src/main/scala/org/ergoplatform/network/ModePeerFeature.scala [hunk 3]
```diff
@@ -15,36 +15,58 @@ import scorex.util.serialization.{Reader, Writer}
...
   * @param blocksToKeep - how many last full blocks the peer is storing
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/network/ModePeerFeature.scala:19]: fixed

**Comment by kushti** [comment]
@aslesarenko comments addressed, please make another pass

**Pull request merged**